### PR TITLE
add tool to delete images from docker registry

### PR DESCRIPTION
**Pull request by planthaber**
Can be improved, but "at least it works"

### CODE REVIEW

**Comment by 0Nel** [comment]
Looks good :+1: Definitely a helpful tools script!

I have not tested it myself as I don't want to delete any images from the registry, but I trust your "at least it works" ;)

Still I have some remarks, before merging:

* Is the ```set -x``` in line 22 a relict from debugging or does it still serve an informational purpose?
* What happens if DIGEST is empty? Will something gets falsely deleted or will the last curl request throw an error? To be sure, I would propose to catch possible curl errors when retrieving the DIGEST variable. Maybe a general ```set -e``` will be enough to abort in case curl fails. However I am not sure if the error from the subshell gets propagated. If not so, maybe squeezing in an if condition checking $? or [ -z $DIGEST ] could be enough to avoid that we request the deletion with an empty DIGEST variable and therefore wrong URL.

On another note, are there any regex rules that could be applied to check the input arguments?

File: tools/delete_image_from_registry.bash [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# helper tool to delete tags from a docker registy
+# Need to run garbage collection afterwards in oder to free disk space https://docs.docker.com/registry/garbage-collection/
+# $> docker exec registry bin/registry garbage-collect --dry-run /etc/docker/registry/config.yml
+
+#https://stackoverflow.com/questions/25436742/how-to-delete-images-from-a-private-docker-registry
+if [$# -ne 4]; then
```
Comment (ErikMichelson) [inline tools/delete_image_from_registry.bash:6]: This is invalid bash syntax and will never fire. The whitespace is important.
```suggestion
if [ $# -ne 4 ]; then
```

**Review by ErikMichelson — CHANGES_REQUESTED** [review]

File: tools/delete_image_from_registry.bash [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ echo "illegal number of parameters"
+ echo use: RegistryUsername registryURL imagename tag
+fi
+
+USER=$1
+REGISTRY=$2
+NAME=$3
+TAG=$4
+read -sp 'registry password:' PASSWD
```
Comment (ErikMichelson) [inline tools/delete_image_from_registry.bash:15]: It's recommended to use the `-r` flag on the read call. When the flag is not set, backslashes are not read as normal characters but rather used for escaping the following sequence. A password usually doesn't have control characters, therefore it's probably better to get the "raw" input.

See also this shellcheck entry: https://github.com/koalaman/shellcheck/wiki/SC2162

File: tools/delete_image_from_registry.bash (hunk 1 above)
Comment (0Nel) [reply tools/delete_image_from_registry.bash:6]: Good catch :+1:

**Comment by planthaber** [comment]
Added the suggested changes apart from digest: the API returns a 404 error if the digest is empty or invalid

**Pull request merged**